docs/source/extend/function-groups.md

```diff
@@ -55,181 +141,843 @@ from nat.builder.function import FunctionGroup
 ```
 
-## Referencing Functions within a Function Group
+**Why use context managers**:
+- Resource lifecycle management (automatic cleanup)
+- Connection pooling efficiency
+- Proper error handling
+- Prevents resource leaks
+
+### Using Configuration Values
 
-Functions are referenced as `instance_name.function_name` (for example, `my.greet`).
+Access configuration values in your functions to customize behavior:
 
 ```python
-async with WorkflowBuilder() as builder:
-    await builder.add_function_group("my", MyGroupConfig(include=["greet"]))
+import httpx
+from nat.cli.register_workflow import register_function_group
 
-    # Able to reference the function directly by its fully qualified name
-    greet = await builder.get_function("my.greet")
-    print(await greet.ainvoke("World"))
+@register_function_group(config_type=APIGroupConfig)
+async def build_api_group(config: APIGroupConfig, _builder: Builder):
+    # Create authenticated HTTP client using config
+    async with httpx.AsyncClient(
+        base_url=config.base_url,
+        headers={"Authorization": f"Bearer {config.api_key}"},
+        timeout=30.0
+    ) as client:
+        group = FunctionGroup(config=config, instance_name="api")
+        
+        async def get_user_fn(user_id: int) -> dict:
+            """Get user details by ID."""
+            response = await client.get(f"/users/{user_id}")
+            response.raise_for_status()
+            return response.json()
+        
+        async def create_item_fn(name: str, description: str) -> dict:
+            """Create a new item."""
+            response = await client.post(
+                "/items",
+                json={"name": name, "description": description}
+            )
+            response.raise_for_status()
+            return response.json()
+        
+        group.add_function(name="get_user", fn=get_user_fn, description=get_user_fn.__doc__)
+        group.add_function(name="create_item", fn=create_item_fn, description=create_item_fn.__doc__)
+        
+        yield group
+```
 
-    my_group = await builder.get_function_group("my")
+## Step 3: Customizing Function Schemas
 
-    # Get all accessible functions in the function group.
-    # If the function group is configured to:
-    # - include some functions, this will return only the included functions.
-    # - not include or exclude any function, this will return all functions in the group.
-    # - exclude some functions, this will return all functions in the group except the excluded functions.
-    accessible_functions = await my_group.get_accessible_functions()
-    
-    # Get all functions in the group.
-    # This will return all functions in the group, regardless of whether they are included or excluded.
-    all_functions = await my_group.get_all_functions()
+The toolkit automatically infers input and output schemas from your function type hints. You can customize these schemas for better validation and documentation.
```

Comment:
Isnt this just the same as normal functions. Should we just link that documentation?